app/src/actions/vesselInfo.js

```diff
@@ -122,9 +121,59 @@ sub/seriesgroup=${seriesGroup}/${i}-01-01T00:00:00.000Z,${i + 1}-01-01T00:00:00.
   };
 }
 
+
+export function addVessel(seriesgroup, series = null, zoomToBounds = false) {
+  return (dispatch) => {
+    dispatch({
+      type: ADD_VESSEL,
+      payload: {
+        seriesgroup
+      }
+    });
+    dispatch(setCurrentVessel(seriesgroup));
+    dispatch(getVesselTrack(seriesgroup, series, zoomToBounds));
+  };
+}
+
 export function clearVesselInfo() {
   return {
-    type: CLEAR_VESSEL_INFO,
-    payload: false
+    type: CLEAR_VESSEL_INFO
+  };
+}
+
+export function toggleActiveVesselPin() {
+  return {
+    type: TOGGLE_ACTIVE_VESSEL_PIN
+  };
+}
+
+export function toggleVesselPin(seriesgroup) {
```

Comment:
I don't like toggle functions, as we are always dependent on hidden previous state, which is not always visible. Can you at least add an optional argument to force a certain state in case we want to?